x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/truncated_label.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiMark } from '@elastic/eui';
+import { EuiHighlight } from '@elastic/eui';
+
+const createContext = () =>
+  document.createElement('canvas').getContext('2d') as CanvasRenderingContext2D;
+
+// extracted from getTextWidth for performance
+const context = createContext();
```

Comment:
This looks like a memory leak, for example if the user navigates away from Lens this object will stay in the DOM.

Reply:
I believe that this will becomes a memory leak only if the element/node gets appended to the DOM.
As long as it's just JS it should be cleared up as any other object by the GC.

Reply:
Agree with @dej611